DependencyInjection/HttplugExtension.php

```diff
@@ -33,5 +36,31 @@ public function load(array $configs, ContainerBuilder $container)
         foreach ($config['main_alias'] as $type => $id) {
             $container->setAlias(sprintf('httplug.%s', $type), $id);
         }
+
+        // Configure services
+        foreach ($config['clients'] as $name => $arguments) {
+            $this->verifyDependencies($arguments['adapter']);
+            $def = $container->register('httplug.'.$name, DummyFactory::class);
+            $def->setFactory([new Reference('httplug.client.factory'), 'getClient'])
```

Comment:
we can't we use the specific factory directly, but need the GenericClientFactory?